src/web/app/pages-instructor/instructor-courses-page/instructor-courses-page.component.ts

```diff
@@ -272,17 +257,16 @@ export class InstructorCoursesPageComponent implements OnInit {
     if (confirm(`Are you sure you want to permanently delete the course: ${courseId}? `
             + 'This operation will delete all students and sessions in this course. '
             + 'All instructors of this course will not be able to access it hereafter as well.')) {
-      const paramMap: { [key: string]: string } = {
-        courseid: courseId,
-        user: this.user,
-      };
-      this.httpRequestService.delete('/instructor/courses/permanentlyDelete', paramMap)
-          .subscribe((resp: MessageOutput) => {
-            this.loadInstructorCourses();
-            this.statusMessageService.showSuccessMessage(resp.message);
-          }, (resp: ErrorMessageOutput) => {
-            this.statusMessageService.showErrorMessage(resp.error.message);
-          });
+      this.courseService.deleteCourse(courseId).subscribe((message: MessageOutput) => {
+        this.loadInstructorCourses();
+        if (message.message === 'OK') {
```

Comment:
This check is redundant as all non-2xx response will be redirect to error callback.

Reply:
Sure, and I've made the necessary changes. However, couldn't there be a case for having this check, as it is safer? In case the API is ever updated without the front-end changing, then if the API ever reports a value other than OK it may not be something the frontend just wants to process as a success.

Reply:
@ChooJeremy there is an example [here](https://github.com/TEAMMATES/teammates/blob/2429ab644d751ff8e2fc310cd6dc854f40992868/src/main/java/teammates/ui/webapi/action/ConfirmFeedbackSessionSubmissionAction.java#L96-L104) for an action with different forms of success. There is clearly only one success case for the action being discussed here, so having an extra "OK" check does not value-add.